packages/server/internal/helpers/asset.go

```diff
@@ -0,0 +1,40 @@
+package helpers
+
+import (
+	"fmt"
+
+	"github.com/alexandros-georgantas/platform-go-challenge/internal/models"
+	"gorm.io/gorm"
+)
+
+func AggregateAssetType(a models.Asset, db gorm.DB) (*models.AssetResponse, error) {
```

Comment:
My initial thought was that I wanted to extract that functionality somewhere as I am using it in a variety of cases, so I wanted something reusable. As I haven’t followed the repository pattern anywhere I thought that placing this under the helpers folder would be fine given the rationale that I already had the pagination helper there which was interacting with the database too. I see now why it is not the most obvious thing to do. As this function is heavily relying on the ORM a better place for it could be inside an AssetRepository where all the interactions with the ORM should be moved. This applies to all my entities